ml-agents-envs/mlagents_envs/environment.py

```diff
@@ -410,15 +414,20 @@ def _close(self, timeout: Optional[int] = None) -> None:
 
     @timed
     def _generate_step_input(
-        self, vector_action: Dict[str, np.ndarray]
+        self, vector_action: Dict[str, ActionTuple]
     ) -> UnityInputProto:
         rl_in = UnityRLInputProto()
         for b in vector_action:
             n_agents = len(self._env_state[b][0])
             if n_agents == 0:
                 continue
             for i in range(n_agents):
-                action = AgentActionProto(vector_actions=vector_action[b][i])
+                # TODO: This check will be removed when the oroto supports hybrid actions
+                if vector_action[b].continuous.shape[1] > 0:
```

Comment:
why check on the shape[1] rather than action_specs.continuous_size?

Reply:
ActionSpec is not available here. This is the only way I think we can determine if the action space is continuous or discrete in this function. **Critically, this is just to support the old proto until the new proto is merged. This is just temporary so that the communication protocol works until the new protos are in place.**